api/app/GraphQL/Mutations/UnarchivePool.php

```diff
@@ -13,8 +13,9 @@ final class UnarchivePool
      */
     public function __invoke($_, array $args)
     {
+        /** @var Pool $pool */
```

Comment:
`find` could return null if the model is not found, right?  Then L18 will throw a null dereference?  Same with unarchive?  Maybe these could be switched to findOrFail?

Reply:
Oh yeah, you are right. For some reason I was assuming this was a `findOrFail`. Fixed in [fix pool type](https://github.com/GCTC-NTGC/gc-digital-talent/pull/11938/commits/bfb90010b18263eeddb892704f9440d779f6d795)